### ISSUE TITLE
Client Shows No Cards or Player Info After A Crash

### ISSUE THREAD

**Issue by Alex-Vasile** [magefree/mage#8687]
My client crashed when I tried playing [[Shiny Impetus]] from exile and paying for it using [[Inspiring Statuary]].

1. I [[Bituminous Blast]] on an opponent's creature
2. From cascade I exile and cast a [[Jeska's Will]]
3. I had both commader on the field so I use both options from Jeska's will
4, Jeska's Will exiled a Shiny Impetus
5. The next turn I try to cast Shiny Impetus 
	- I have an Inspiring Statuary on the field, so I am going to tap some of my treason tokens with it to pay for Shiny Impetus.
	- The first time I try to pay, I mistakenly use Inspiring Statuary's ability first to pay for the {2} before paying for the {R}
	- I hit cancel. Then I try casting it again by tapping a Mountain for {R} first.
	- When I try to pay the cost the client crashes. I don't remember if it crashed on the paying of the {R} or once the I try paying with Inspiring Statuary's ability.

When I reconnect I can't see anything on the board. None of the cards, not even the rectangles containing play names, mana in pool, etc. Resizing the window does nothing to bring them back, continue on does not either.

On the board:
- Mountain and Swamps
- 4-5 treasure tokens
- 3 ramp artifacts

Running on a mac.

The client log does not seem to have anything written in it at the moment that it crashed. But it does seem to have information about the UI bug afterwards

The client log has:
```
INFO  2022-02-10 20:49:09,741 Starting MAGE client version 1.4.51-V3-beta4 (build: 2022-02-06 06:11)                     =>[main] MageFrame.main 
INFO  2022-02-10 20:49:09,743 Logging level: INFO                                                                        =>[main] MageFrame.main 
INFO  2022-02-10 20:49:09,743 Default charset: UTF-8                                                                     =>[main] MageFrame.main 
INFO  2022-02-10 20:49:12,137 Using GUI theme: Default                                                                   =>[AWT-EventQueue-0] PreferencesDialog.getCurrentTheme 
INFO  2022-02-10 20:49:12,159 Loading database...                                                                        =>[AWT-EventQueue-0] RepositoryUtil.bootstrapLocalDb 
INFO  2022-02-10 20:49:13,154 Images: search broken files...                                                             =>[AWT-EventQueue-0] MageFrame.<init> 
INFO  2022-02-10 20:49:13,473 Loading symbols...                                                                         =>[AWT-EventQueue-0] ManaSymbols.loadImages 
INFO  2022-02-10 20:49:15,620 Loading plugins...                                                                         =>[AWT-EventQueue-0] Plugins.loadPlugins 
INFO  2022-02-10 20:49:15,684 [Game counter plugin, version 0.1] has been loaded.                                        =>[AWT-EventQueue-0] CounterPluginImpl.newPlugin 
INFO  2022-02-10 20:49:15,694 [Card plugin, version 0.7] has been loaded.                                                =>[AWT-EventQueue-0] CardPluginImpl.newPlugin 
INFO  2022-02-10 20:49:15,697 [Theme plugin, version 0.5] has been loaded.                                               =>[AWT-EventQueue-0] ThemePluginImpl.newPlugin 
INFO  2022-02-10 20:49:15,697 Done.                                                                                      =>[AWT-EventQueue-0] Plugins.loadPlugins 
INFO  2022-02-10 20:49:17,329 Client start up time: 7 seconds                                                            =>[AWT-EventQueue-0] MageFrame.lambda$new$2 
INFO  2022-02-10 21:00:28,279 Connect: connecting to server localhost:17171                                              =>[SwingWorker-pool-5-thread-4] SessionImpl$5.work 
INFO  2022-02-10 21:00:28,618 Connect: DONE                                                                              =>[SwingWorker-pool-5-thread-4] SessionImpl$5.work 
INFO  2022-02-10 21:00:28,620 Logging: as username b to server localhost:17171                                           =>[SwingWorker-pool-5-thread-5] SessionImpl$4.work 
INFO  2022-02-10 21:00:28,765 Logging: DONE                                                                              =>[SwingWorker-pool-5-thread-5] SessionImpl$4.work 
INFO  2022-02-10 21:01:46,651 Game fa3a1634-4266-4655-bd42-3689fbaec994 started for player a81108f9-3a3c-490d-b9f6-22d0200501b0 =>[AWT-EventQueue-0] CallbackClientImpl.gameStarted 
WARN  2022-02-10 21:03:09,661 catch un-synced message from later source (possible reason: connection or performance problems): 134, text=Draw - Waiting for <font color='#20B2AA'>Computer 2</font> =>[AWT-EventQueue-0] FeedbackPanel.prepareFeedback 
WARN  2022-02-10 21:05:00,917 catch un-synced message from later source (possible reason: connection or performance problems): 210, text=Declare Attackers - Waiting for <font color='#20B2AA'>Computer 2</font> =>[AWT-EventQueue-0] FeedbackPanel.prepareFeedback 
WARN  2022-02-10 21:14:56,188 catch un-synced message from later source (possible reason: connection or performance problems): 676, text=Declare Attackers - Waiting for <font color='#20B2AA'>Computer 2</font> =>[AWT-EventQueue-0] FeedbackPanel.prepareFeedback 
WARN  2022-02-10 21:15:20,965 catch un-synced message from later source (possible reason: connection or performance problems): 690, text=End Combat - Waiting for <font color='#20B2AA'>Computer 2</font> =>[AWT-EventQueue-0] FeedbackPanel.prepareFeedback 
INFO  2022-02-10 21:19:27,562 Starting MAGE client version 1.4.51-V3-beta4 (build: 2022-02-06 06:11)                     =>[main] MageFrame.main 
INFO  2022-02-10 21:19:27,564 Logging level: INFO                                                                        =>[main] MageFrame.main 
INFO  2022-02-10 21:19:27,564 Default charset: UTF-8                                                                     =>[main] MageFrame.main 
INFO  2022-02-10 21:19:30,043 Using GUI theme: Default                                                                   =>[AWT-EventQueue-0] PreferencesDialog.getCurrentTheme 
INFO  2022-02-10 21:19:30,070 Loading database...                                                                        =>[AWT-EventQueue-0] RepositoryUtil.bootstrapLocalDb 
INFO  2022-02-10 21:19:34,858 Images: search broken files...                                                             =>[AWT-EventQueue-0] MageFrame.<init> 
WARN  2022-02-10 21:19:34,926 Images: found 30 broken files. Trying to fix it...                                         =>[AWT-EventQueue-0] CardImageUtils.checkAndFixImageFiles 
INFO  2022-02-10 21:19:35,244 Loading symbols...                                                                         =>[AWT-EventQueue-0] ManaSymbols.loadImages 
INFO  2022-02-10 21:19:36,606 Loading plugins...                                                                         =>[AWT-EventQueue-0] Plugins.loadPlugins 
INFO  2022-02-10 21:19:36,698 [Game counter plugin, version 0.1] has been loaded.                                        =>[AWT-EventQueue-0] CounterPluginImpl.newPlugin 
INFO  2022-02-10 21:19:36,708 [Card plugin, version 0.7] has been loaded.                                                =>[AWT-EventQueue-0] CardPluginImpl.newPlugin 
INFO  2022-02-10 21:19:36,711 [Theme plugin, version 0.5] has been loaded.                                               =>[AWT-EventQueue-0] ThemePluginImpl.newPlugin 
INFO  2022-02-10 21:19:36,711 Done.                                                                                      =>[AWT-EventQueue-0] Plugins.loadPlugins 
INFO  2022-02-10 21:19:38,782 Client start up time: 11 seconds                                                           =>[AWT-EventQueue-0] MageFrame.lambda$new$2 
INFO  2022-02-10 21:19:40,643 Connect: connecting to server localhost:17171                                              =>[SwingWorker-pool-5-thread-4] SessionImpl$5.work 
INFO  2022-02-10 21:19:40,916 Connect: DONE                                                                              =>[SwingWorker-pool-5-thread-4] SessionImpl$5.work 
INFO  2022-02-10 21:19:40,920 Logging: as username b to server localhost:17171                                           =>[SwingWorker-pool-5-thread-5] SessionImpl$4.work 
INFO  2022-02-10 21:19:41,134 Logging: DONE                                                                              =>[SwingWorker-pool-5-thread-5] SessionImpl$4.work 
INFO  2022-02-10 21:19:48,833 Game fa3a1634-4266-4655-bd42-3689fbaec994 started for player a81108f9-3a3c-490d-b9f6-22d0200501b0 =>[AWT-EventQueue-0] CallbackClientImpl.gameStarted 
FATAL 2022-02-10 21:21:18,845 Client error
                                                                              =>[AWT-EventQueue-0] CallbackClientImpl.handleException 
java.lang.NullPointerException
	at mage.client.game.GamePanel.updateGame(GamePanel.java:911)
	at mage.client.game.GamePanel.updateGame(GamePanel.java:713)
	at mage.client.remote.CallbackClientImpl.lambda$processCallback$0(CallbackClientImpl.java:311)
	at java.awt.event.InvocationEvent.dispatch(InvocationEvent.java:311)
	at java.awt.EventQueue.dispatchEventImpl(EventQueue.java:758)
	at java.awt.EventQueue.access$500(EventQueue.java:97)
	at java.awt.EventQueue$3.run(EventQueue.java:709)
	at java.awt.EventQueue$3.run(EventQueue.java:703)
	at java.security.AccessController.doPrivileged(Native Method)
	at java.security.ProtectionDomain$JavaSecurityAccessImpl.doIntersectionPrivilege(ProtectionDomain.java:74)
	at java.awt.EventQueue.dispatchEvent(EventQueue.java:728)
	at java.awt.EventDispatchThread.pumpOneEventForFilters(EventDispatchThread.java:205)
	at java.awt.EventDispatchThread.pumpEventsForFilter(EventDispatchThread.java:116)
	at java.awt.EventDispatchThread.pumpEventsForHierarchy(EventDispatchThread.java:105)
	at java.awt.EventDispatchThread.pumpEvents(EventDispatchThread.java:101)
	at java.awt.EventDispatchThread.pumpEvents(EventDispatchThread.java:93)
	at java.awt.EventDispatchThread.run(EventDispatchThread.java:82)
FATAL 2022-02-10 21:21:18,975 Client error
                                                                              =>[AWT-EventQueue-0] CallbackClientImpl.handleException 
java.lang.NullPointerException
	at mage.client.game.GamePanel.updateGame(GamePanel.java:911)
	at mage.client.game.GamePanel.updateGame(GamePanel.java:713)
	at mage.client.game.GamePanel.endMessage(GamePanel.java:1664)
	at mage.client.remote.CallbackClientImpl.lambda$processCallback$0(CallbackClientImpl.java:193)
	at java.awt.event.InvocationEvent.dispatch(InvocationEvent.java:311)
	at mage.client.dialog.MageDialog.startModal(MageDialog.java:172)
	at mage.client.dialog.MageDialog.show(MageDialog.java:83)
	at java.awt.Component.show(Component.java:1671)
	at java.awt.Component.setVisible(Component.java:1623)
	at javax.swing.JComponent.setVisible(JComponent.java:2644)
	at mage.client.dialog.MageDialog.setVisible(MageDialog.java:89)
	at mage.client.dialog.UserRequestDialog.showDialog(UserRequestDialog.java:78)
	at mage.client.MageFrame.showUserRequestDialog(MageFrame.java:1238)
	at mage.client.MageFrame.showError(MageFrame.java:1512)
	at mage.client.remote.CallbackClientImpl.handleException(CallbackClientImpl.java:633)
	at mage.client.remote.CallbackClientImpl.lambda$processCallback$0(CallbackClientImpl.java:444)
	at java.awt.event.InvocationEvent.dispatch(InvocationEvent.java:311)
	at java.awt.EventQueue.dispatchEventImpl(EventQueue.java:758)
	at java.awt.EventQueue.access$500(EventQueue.java:97)
	at java.awt.EventQueue$3.run(EventQueue.java:709)
	at java.awt.EventQueue$3.run(EventQueue.java:703)
	at java.security.AccessController.doPrivileged(Native Method)
	at java.security.ProtectionDomain$JavaSecurityAccessImpl.doIntersectionPrivilege(ProtectionDomain.java:74)
	at java.awt.EventQueue.dispatchEvent(EventQueue.java:728)
	at java.awt.EventDispatchThread.pumpOneEventForFilters(EventDispatchThread.java:205)
	at java.awt.EventDispatchThread.pumpEventsForFilter(EventDispatchThread.java:116)
	at java.awt.EventDispatchThread.pumpEventsForHierarchy(EventDispatchThread.java:105)
	at java.awt.EventDispatchThread.pumpEvents(EventDispatchThread.java:101)
	at java.awt.EventDispatchThread.pumpEvents(EventDispatchThread.java:93)
	at java.awt.EventDispatchThread.run(EventDispatchThread.java:82)
```

The server log has:
```
INFO  2022-02-10 20:58:52,990 Started MAGE server - listening on 0.0.0.0:17171/?serializationtype=jboss&maxPoolSize=300  =>[main] Main.main 
INFO  2022-02-10 21:00:28,654 b connected to server                                                                      =>[WorkerThread#0[192.168.1.66:56723]] SessionManagerImpl.connectUser 
INFO  2022-02-10 21:01:45,880 MATCH started [Game] 0e56ff04-46af-4a93-9dde-ee695fa57691(b)                               =>[CALL main-40] TableController.lambda$startMatch$7 
INFO  2022-02-10 21:01:45,982 GAME started fa3a1634-4266-4655-bd42-3689fbaec994 [Game] b - Computer 2                    =>[CALL main-40] TableController.startGame 
INFO  2022-02-10 21:19:28,981 LOST CONNECTION - b [Match: 1 ] at 192.168.1.66 sessionId: 5c4o11u-nhjwf3-kzhrj2qq-1-kzhrj2ss-4 =>[Timer-0] Main$ClientConnectionListener.handleConnectionException 
INFO  2022-02-10 21:19:40,931 Reconnecting session for b                                                                 =>[WorkerThread#3[192.168.1.66:56974]] Session.connectUserHandling 
INFO  2022-02-10 21:19:41,008 b connected to server                                                                      =>[WorkerThread#3[192.168.1.66:56974]] SessionManagerImpl.connectUser 
INFO  2022-02-10 21:21:50,869 LOST CONNECTION - b [] at 192.168.1.66 sessionId: 5c4o11u-1fi94w-kzhs7rwo-1-kzhs7ryk-4     =>[Timer-0] Main$ClientConnectionListener.handleConnectionException 
INFO  2022-02-10 21:24:49,762 USER REMOVE - b (SessionExpired)  userId: 9bbe2050-8713-4425-82aa-b94decd90c28 []          =>[CALL main-68] UserManagerImpl.lambda$null$1 
```

**Comment by Alex-Vasile** [comment]
I have tried reproducing the first issue (client crashing) myself, and was not able to using the steps above.

The client did crash a few more times randomly and different spots. I think it may be a duplicate of #8534.

I have also not had luck reproducing the second error (reconnecting and nothing being shown for the hands, battlefield, or players. But that seems to be a separate issue.

I've seen this message a few times `Failed to write core dump. Core dumps have been disabled. To enable core dumping, try "ulimit -c unlimited" before starting Java again`.

I googled around a bit, and it seems like others recommend updating the java jre to one that is built for arm natively, but I can't do that until #8627 is merged in.

**Comment by JayDi85** [comment]
Empty battlefield related to non serializable object in some of the cards/effects (will see server errors) or to wrong client/server versions (incompatible java classes for serializable) or to client side error (e.g. with images/connection/graphic).

> reconnecting and nothing being shown for the hands, battlefield, or players

Known issue related to miss game started event for the client (client don't get it in some random use cases after reconnect).

**Comment by Alex-Vasile** [comment]
> Empty battlefield related to non serializable object in some of the cards/effects (will see server errors) or to wrong client/server versions (incompatible java classes for serializable) or to client side error (e.g. with images/connection/graphic).
> 
> > reconnecting and nothing being shown for the hands, battlefield, or players
> 
> Known issue related to miss game started event for the client (client don't get it in some random use cases after reconnect).

I see. So this is a duplicate of another issue?

**Comment by JayDi85** [comment]
> java.lang.NullPointerException at mage.client.game.GamePanel.updateGame(GamePanel.java:911)

It's related to connection problem (client disconnected, e.g. GUI freezes in infinite loop or lost a ping session) -- in most use cases you must see disconnected message, but somehow it can be miss with such error type.

> So this is a duplicate of another issue

Nope (see above about disconnection). If it's unproduceably then can be closed.
